TestResultSummaryService/parsers/BenchmarkParser.js

```diff
@@ -114,20 +114,29 @@ class BenchmarkParser extends Parser {
                 
                 // Parse metric values
                 curTestData["metrics"] = [];
-                for ( let i = 0; i < curBenchVariant["metrics"].length; i++ ) {
+                let curMetricList = Object.keys(curBenchVariant["metrics"]);
+                for ( let i = 0; i < curMetricList.length; i++ ) {
 
-                    curMetric = curBenchVariant["metrics"][i];  
+                    curMetric = curMetricList[i];
+                    curRegex = curBenchVariant["metrics"][curMetric]["regex"];  
                     /*	Parse all values for single metric from result to an array
                      *  e.g 
                      *  Liberty Startup =>
                      *  curRegexResult = ['startup time in ms 32',32,'startup time in ms 41',41, x 6] 
                      *  Liberty Throughput =>
                      *  curRegexResult = ['<metric type="throughput">32<\/data>',32]
                      */
-                    curRegexResult = curSearchString.split(curMetric.regex);
+                    curRegexResult = curSearchString.split(curRegex);
                     //collect only the capture groups from the regex array
-                    curMetricValue = curRegexResult.filter ( (value,index) => (index %2 == 1) ? parseFloat(value):null);
-                    curTestData["metrics"].push({name: curMetric.name, value: curMetricValue});
+                    curRegexResult = curRegexResult.filter( (value,index) => (index %2 == 1));
+                    curMetricValue = curRegexResult.map(parseFloat);
+                    //For metric such as JITCPU total, you need to sum up the values
+                    if(curBenchVariant[curMetric]["sum"] != undefined) {
+                        console.log(curMetricValue);
+                        curMetricValue = curMetricValue.reduce(add);
+                        console.log(curMetricValue);
+                    }
+                    curTestData["metrics"].push({name: curMetric, value: curMetricValue});
```

Comment:
Let's soft-code this by adding something like `funcName` to any metric that needs some special processing. Let's also update the doc on top of the parser for its purpose. That `funcName` could have something like `math.sum`.